glib/src/subclass/types.rs

```diff
@@ -233,23 +234,60 @@ impl TypeData {
         self.parent_class
     }
 
-    /// Returns a pointer to the interface implementation specific data.
+    /// Returns a pointer to the class implementation specific data.
     ///
-    /// This is used for interface implementations to store additional data.
-    pub fn get_interface_data(&self, type_: ffi::GType) -> ffi::gpointer {
+    /// This is used for class implementations to store additional data.
+    pub fn get_class_data<T: Any + Send + Sync + 'static>(&self, type_: ffi::GType) -> Option<&T> {
         unsafe {
-            if self.interface_data.is_null() {
-                return ptr::null_mut();
+            if self.class_data.is_none() {
+                return None;
             }
+            let map = self.class_data.unwrap();
+            map.as_ref().get(&type_).map(|v| v.downcast_ref()).flatten()
```

Comment:
This doesn't compile due to 
```

   --> glib/src/subclass/types.rs:246:13
    |
246 |             map.as_ref().get(&type_).map(|v| v.downcast_ref()).flatten()
    |             ---^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             returns a value referencing data owned by the current function
```
not sure how to handle this?

Reply:
Use a `match` instead of `as_ref()` and `map()` and `flatten()`. Also your `map(f).flatten()` is basically `and_then(f)` :)

Reply:
The as_ref here comes from NonNull and not Option

Reply:
Thanks for the tip on and_then, wasn't aware of that :)

Reply:
This still doesn't compile btw :P